### PR TITLE
tweak INTERNAL_TEST_NUMBER sending

### PR DESCRIPTION
**Pull request by sastels**
# Summary | Résumé

When we send to the INTERNAL_TEST_NUMBER we were queueing the fake callback before setting the reference in the notification. This could lead to the callback failing on its first attempt.

Here we set the reference before creating the fake callback.

# Test instructions | Instructions pour tester la modification

Test on staging - make sure SMS to this number get marked delivered.

# Release Instructions | Instructions pour le déploiement

None.

# Reviewer checklist | Liste de vérification du réviseur

- [ ] This PR does not break existing functionality.
- [ ] This PR does not violate GCNotify's privacy policies.
- [ ] This PR does not raise new security concerns. Refer to our GC Notify Risk Register document on our Google drive.
- [ ] This PR does not significantly alter performance.
- [ ] Additional required documentation resulting of these changes is covered (such as the README, setup instructions, a related ADR or the technical documentation).

> ⚠ If boxes cannot be checked off before merging the PR, they should be moved to the "Release Instructions" section with appropriate steps required to verify before release. For example, changes to celery code may require tests on staging to verify that performance has not been affected.

### CODE REVIEW

File: tests/app/delivery/test_send_to_providers.py [hunk 1]
```diff
@@ -490,29 +488,19 @@ def test_should_call_send_sms_response_task_if_research_mode(
...
     assert persisted_notification.to == sample_notification.to
     assert persisted_notification.template_id == sample_notification.template_id
     assert persisted_notification.status == "sent"
     assert persisted_notification.sent_at <= datetime.utcnow()
     assert persisted_notification.sent_by == "sns"
-    assert persisted_notification.reference == reference
     assert not persisted_notification.personalisation
 
 
-def test_should_not_have_sent_status_if_fake_callback_function_fails(sample_notification, mocker):
```
Comment (sastels) [inline tests/app/delivery/test_send_to_providers.py:505]: This test was for the corner case where a user would send with a TEST key and somehow our fake callback code would fail (so the sms would stay CREATED and get resent). We're explicitly marking it SENT first now to make the callbacks more likely to pass the first time so this test no longer works.

**Review by ben851 — APPROVED** [review]
I think so?
